l10n_it_withholding_tax_payment/models/withholding_tax.py

```diff
@@ -0,0 +1,184 @@
+    account_id = fields.Many2one('account.account', string='Account')
+    journal_id = fields.Many2one('account.journal', string='Journal')
+    line_ids = fields.One2many(
+        'withholding.tax.move', 'wt_move_payment_id', string='Lines')
+    amount = fields.Float('WT amount', compute='_compute_total')
+
+    def create_account_move(self):
+        account_move_obj = self.env['account.move']
+        for mp in self:
+            if not mp.date_payment \
+                    or not mp.journal_id\
+                    or not mp.account_id:
+                raise ValidationError(
+                    _("Warning! Datas required for account move creation:"
+                      " payment date, journal, account")
+                )
+            # WT Moves
+            wt_move_balance = 0
+            move_lines = []
+            for wt_move in mp.line_ids:
+                debit = 0
+                credit = 0
+                if wt_move.amount > 0:
+                    debit = wt_move.amount
+                else:
+                    credit = wt_move.amount
+                vals = {
+                    'name': _('Withholding Tax Payment %s')
+                    % wt_move.partner_id.name,
+                    'account_id':
+                        wt_move.withholding_tax_id.account_payable_id.id,
+                    'credit': credit,
+                    'debit': debit,
+                }
+                move_lines.append((0, 0, vals))
+                # Balance
+                wt_move_balance += wt_move.amount
+            # WT payment
+            if wt_move_balance:
+                debit = 0
+                credit = 0
+                if wt_move_balance > 0:
+                    credit = wt_move_balance
+                else:
+                    debit = wt_move_balance * -1
+                vals = {
+                    'name': _('Withholding Tax Payment'),
+                    'account_id': mp.account_id.id,
+                    'credit': credit,
+                    'debit': debit,
+                }
+                move_lines.append((0, 0, vals))
+            # Move create
+            move = account_move_obj.create([{
+                'ref': _('Withholding Tax Payment'),
+                'journal_id': mp.journal_id.id,
+                'date': mp.date_payment,
+                'line_ids': move_lines,
+            }])
+            move.post()
+            # Ref on payement
+            mp.move_id = move.id
+
+    def generate_from_moves(self, wt_moves):
+        # Moves must have the same company
+        if wt_moves and len(wt_moves.mapped('company_id')) > 1:
+            raise ValidationError(
+                _("The selected moves must have the same company!"))
+        sequence_obj = self.env['ir.sequence']
+        for wt_move in wt_moves:
+            if wt_move.state == 'paid':
+                raise ValidationError(
+                    _("Wt move already paid! - %s - %s - %s")
```

Comment:
Dato che un acronimo lo metterei in maiuscolo (anche per uniformità).
s/Wt/WT

Compare anche in altre stringhe.

Reply:
Fatto! 👌